File: RefugeUA.Entities/VolunteerEvent.cs
using RefugeUA.Entities.Abstracts;
using RefugeUA.Entities.Enums;
using RefugeUA.Entities.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RefugeUA.Entities
{
    public class VolunteerEvent : Entity
    {
        public VolunteerEvent()
        {
            this.CreatedAt = DateTime.Now;
        }

        public string Title { get; set; } = default!;

        public string Content { get; set; } = default!;

        public DateTime? StartTime { get; set; }

        public DateTime? EndTime { get; set; }

        public DateTime CreatedAt { get; set; }

        public ICollection<IUser> Participants { get; set; } = default!;

        public ICollection<IUser> Organizers { get; set; } = default!;

        public ICollection<VolunteerEventScheduleItem>? ScheduleItems { get; set; }

        public long? AddressId { get; set; }

        public Address? Address { get; set; } = default!;

        public long? VolunteerGroupId { get; set; }

        public VolunteerGroup VolunteerGroup { get; set; } = default!;

        public string? OnlineConferenceLink { get; set; }

        public bool IsClosed { get; set; }

        public VolunteerEventType EventType { get; set; }

        public string? DonationLink { get; set; }
    }
}

File: RefugeUA.WebApp/refugeua.webapp.client/src/app/core/models/volunteer-event.ts
import { Entity } from "./entity"
import { Address } from "./address"
import { User } from "./user"
import { VolunteerGroup } from "./volunteer-group"
import { VolunteerEventScheduleItem } from "./volunteer-event-schedule-item"
import { VolunteerEventType } from "../enums/volunteer-event-type-enum"

export interface VolunteerEvent extends Entity {
    title: string,
    content: string,
    startTime: Date,
    endTime: Date,
    createdAt: Date,
    addressId?: number,
    address?: Address,
    scheduleItems?: VolunteerEventScheduleItem[],
    volunteerGroupId?: number,
    volunteerGroup?: VolunteerGroup,
    volunteerGroupTitle?: string,
    onlineConferenceLink?: string,
    isClosed: boolean,
    donationLink?: string,
    volunteerEventType: VolunteerEventType,
    organizers: User[],
    participants?: User[],
    participantsCount?: number
}